src/main/java/net/logstash/logback/marker/MapEntriesAppendingMarker.java

```diff
@@ -99,13 +99,7 @@ public String toStringSelf() {
 
     @Override
     public boolean equals(Object obj) {
-        if (this == obj) {
-            return true;
-        }
```

Comment:
I think the == check should remain here (and in the other equals methods) as a short-circuit optimization to handle the case where the same object is being compared to itself.

Reply:
This check is still there - but handled by `super.equals()`...
... or do you mean you'd prefer to keep the `==` to avoid an extra call to super() ?

Reply:
I'd prefer to keep the `==` check here to avoid executing `Objects.equals(this.map, other.map)` when the object is being compared to itself.

The `==` check in `super.equals(obj)` does not prevent the map comparison when the object is being compared to itself.

Reply:
Arghh.. yes indeed, you are right.. I missed that :-(
Let me revert these changes!